### PR TITLE
COG-1468 Notification Web Service to Consume Registration Complete event

### PR DESCRIPTION
**Pull request by sorych**
### JIRA Issue Link
- [COG-1468 Notification Web Service to Consume Registration Complete event](https://osi-cwds.atlassian.net/browse/COG-1468)

### Tests
- [x] I have included unit tests 
- [x] I have included integration tests 
- [ ] I have included performance tests 
- [ ] I have included other tests 
- [ ] I have NOT included tests 

### Types of changes

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Checklist:

- [ ] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build, to check Lint/Rubocop, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: src/test/resources/fixtures/idm/notify/invalid.json [hunk 1]
```diff
@@ -1,4 +1,4 @@
 {
-  "userId": "17067e4e-270f-4623-b86c-b4d4fa527a34",
-  "actionType": "invalidOperation"
+  "user_id": "17067e4e-270f-4623-b86c-b4d4fa527a34",
```
Comment (dmytro-palczewski-osi) [inline src/test/resources/fixtures/idm/notify/invalid.json]: In my opinion properties style is a part of external contract. So we cannot change these test resource.

File: src/main/java/gov/ca/cwds/idm/dto/IdmUserNotification.java [hunk 2]
```diff
@@ -15,12 +13,7 @@
   private String userId;
   private String actionType;
 
-  @JsonCreator
-  public IdmUserNotification(
-      @JsonProperty("userId") String userId,
-      @JsonProperty("actionType") String actionType) {
-    this.userId = userId;
-    this.actionType = actionType;
+  public IdmUserNotification() {
```
Comment (dmytro-palczewski-osi) [inline src/main/java/gov/ca/cwds/idm/dto/IdmUserNotification.java]: We need this constructor to support camel-styled properties in external events.

File: src/test/resources/fixtures/idm/notify/invalid.json (hunk 1 above)
Comment (sorych) [reply src/test/resources/fixtures/idm/notify/invalid.json]: what do you mean external contract?
we are establishing the contract here and it should correspond to our common style, snake case

File: src/main/java/gov/ca/cwds/idm/dto/IdmUserNotification.java (hunk 2 above)
Comment (sorych) [reply src/main/java/gov/ca/cwds/idm/dto/IdmUserNotification.java]: it can be done both ways, I just used other

**Pull request merged**